apps/server/src/modules/user-login-migration/service/common-user-login-migration.service.spec.ts

```diff
@@ -0,0 +1,161 @@
+
+	let authorizationService: DeepMocked<AuthorizationService>;
+	let userLoginMigration: DeepMocked<UserLoginMigrationService>;
+	let schoolService: DeepMocked<SchoolService>;
+
+	beforeAll(async () => {
+		await setupEntities();
+
+		module = await Test.createTestingModule({
+			providers: [
+				CommonUserLoginMigrationService,
+				{
+					provide: AuthorizationService,
+					useValue: createMock<AuthorizationService>(),
+				},
+				{
+					provide: UserLoginMigrationService,
+					useValue: createMock<UserLoginMigrationService>(),
+				},
+				{
+					provide: SchoolService,
+					useValue: createMock<SchoolService>(),
+				},
+			],
+		}).compile();
+
+		service = module.get(CommonUserLoginMigrationService);
+		authorizationService = module.get(AuthorizationService);
+		userLoginMigration = module.get(UserLoginMigrationService);
+		schoolService = module.get(SchoolService);
+	});
+
+	afterAll(async () => {
+		await module.close();
+	});
+
+	afterEach(() => {
+		jest.clearAllMocks();
+	});
+
+	describe('ensurePermission is called', () => {
+		describe('when context, user and school are given ', () => {
+			const setup = () => {
+				const adminRole = roleFactory.buildWithId({
+					name: RoleName.ADMINISTRATOR,
+					permissions: [Permission.USER_LOGIN_MIGRATION_ADMIN],
+				});
+
+				const school = schoolDOFactory.buildWithId();
+				const user: User = userFactory.buildWithId();
+				const context: AuthorizationContext = {
+					action: Action.write,
+					requiredPermissions: [Permission.USER_LOGIN_MIGRATION_ADMIN],
+				};
+				authorizationService.getUserWithPermissions.mockResolvedValue(user);
+				schoolService.getSchoolById.mockResolvedValue(school);
+				authorizationService.checkPermissionByReferences.mockResolvedValue(Promise.resolve());
+
+				return {
+					user,
+					schoolId: school.id as string,
+					adminRole,
+					context,
+					school,
+				};
+			};
+
+			it('should call getUserWithPermissions', async () => {
+				const { user, schoolId } = setup();
+
+				await service.ensurePermission(user.id, schoolId);
+
+				expect(authorizationService.getUserWithPermissions).toHaveBeenCalledWith(user.id);
+			});
+
+			it('should call getSchoolById', async () => {
+				const { user, schoolId } = setup();
+
+				await service.ensurePermission(user.id, schoolId);
+
+				expect(schoolService.getSchoolById).toHaveBeenCalledWith(schoolId);
+			});
+
+			it('should call checkPermissionByReferences', async () => {
+				const { user, school, context } = setup();
+
+				await service.ensurePermission(user.id, school.id as string);
+
+				expect(authorizationService.checkPermission).toHaveBeenCalledWith(user, school, context);
+			});
+		});
+	});
+
+	describe('findExistingUserLoginMigration is called', () => {
+		describe('when context, user and school are given ', () => {
+			const setup = () => {
+				const userId = 'userId';
+
+				const migration: UserLoginMigrationDO = new UserLoginMigrationDO({
```

Comment:
A testFactory should be already exists can you please use it for all the test in the PR?

Reply:
I refactored this also in other test files.